weblab/entities/processing.py

```diff
@@ -171,21 +170,24 @@ def record_experiments_to_run(user, entity, commit):
         entity.entity_type: entity,
         entity.entity_type + '_version': commit.sha,
     }
-    for parent in commit.parents:
-        # Find visible experiments involving this parent
-        parent_kwargs = {
-            entity.entity_type: entity,
-            entity.entity_type + '_version': entity.repocache.get_version(parent.sha),
-        }
-        for expt in Experiment.objects.filter(**parent_kwargs):
-            if expt.is_visible_to_user(user):
+    if entity.other_type == entity.ENTITY_TYPE_MODEL:
+        other_model = ModelEntity
+    else:
+        other_model = ProtocolEntity
+    search_args = {
+        entity.other_type + '_experiments__' + entity.entity_type: entity,
+    }
+    other_entities = other_model.objects.visible_to_user(user).filter(**search_args)
+    for other_entity in other_entities:
+        # Look for latest visible version
+        for other_version in other_entity.cachedentity.versions.reverse():
+            if other_entity.is_version_visible_to_user(other_version.sha, user):
                 # Record the new experiment to run
                 kwargs = {
                     'submitter': user,
-                    'model_id': expt.model_id,
-                    'model_version': expt.model_version.sha,
-                    'protocol_id': expt.protocol_id,
-                    'protocol_version': expt.protocol_version.sha,
+                    entity.other_type: other_entity,
+                    other_entity + '_version': other_version.sha,
```

Comment:
This line errors with `TypeError: unsupported operand type(s) for +: 'ProtocolEntity' and 'str'`. I wonder if the fix is as simple as wrapping other_entity in str()?

Reply:
Ah, no, I think it should be `entity.other_type + '_version'` sorry!